ethtxmanager/ethtxmanager.go

```diff
@@ -615,8 +615,8 @@ func (c *Client) monitorTx(ctx context.Context, mTx *monitoredTxnIteration, logg
 		logger.Debugf("signed tx %v created", signedTx.Hash().String())
 
 		// add tx to monitored tx history
-		err = mTx.AddHistory(signedTx)
-		if errors.Is(err, ErrAlreadyExists) {
+		found, err := mTx.AddHistory(signedTx)
+		if found {
```

Comment:
I would say that the previous check was good as well. Maybe the crucial part is that you have changed the function receiver to the pointer in the `AddHistory`.

I have even changed the test to make sure it checks against the `ErrAlreadyExists` and it works as expected (eebb1485d8d064fa3a5419b8a58a822c46ed4260)

WDYT?

Reply:
During my testing errors.Is and err == ErrAlreadyExists didn't work. Also changing the receiver to a pointer didn't fix the issue. It only worked properly when checking the boolean.

The change in the test looks ok to me.

Reply:
WDYT about this snippet, would it work (it still seems to me that the boolean in the `AddHistory` is redundant)?

```go
// add tx to monitored tx history
_, err := mTx.AddHistory(signedTx)
if err != nil {
	if errors.Is(err, ErrAlreadyExists) {
		logger.Infof("signed tx already existed in the history")
	} else {
		logger.Errorf("failed to add signed tx %v to monitored tx history: %v", signedTx.Hash().String(), err)
		return
	}
}

// update monitored tx changes into storage
err = c.storage.Update(ctx, *mTx.MonitoredTx)
if err != nil {
	logger.Errorf("failed to update monitored tx: %v", err)
	return
}
logger.Debugf("signed tx added to the monitored tx history")
```

Reply:
This check wold also fail, as it is the same we had but now inner to the other if:

`if errors.Is(err, ErrAlreadyExists) {`